lib/grpc_reflection/service/agent.ex

```diff
@@ -8,16 +8,18 @@ defmodule GrpcReflection.Service.Agent do
   alias GrpcReflection.Service.Builder
   alias GrpcReflection.Service.Lookup
 
-  defstruct services: [], files: %{}, symbols: %{}
+  defstruct services: [], files: %{}, symbols: %{}, extensions: %{}
 
   @type descriptor_t :: GrpcReflection.descriptor_t()
   @type t :: %{
           required(:services) => list(module()),
           required(:files) => %{optional(binary()) => descriptor_t()},
-          required(:symbols) => %{optional(binary()) => descriptor_t()}
+          required(:symbols) => %{optional(binary()) => descriptor_t()},
+          required(:extensions) => %{optional(binary()) => list(integer())}
         }
 
   def start_link(_, opts) do
+    Protobuf.load_extensions()
```

Comment:
I don't think it is most effective to call `Protobuf.load_extensions()` here.  Inside of `put_state` the state comes into the Agent already calculated, and then we lazy-init the Agent by calling `start_agent_on_first_call`.

I think the intent is that this is called in the BEAM _before_ we calculate the reflection state so that extensions are available.  This may not be the case here.

There is a discussion [here](https://elixirforum.com/t/idiomatic-place-for-this-code-when-the-application-boots/34395) that covers some of the ways to always initialize one-time tasking.  I;ve used the "returning :ignore" approach without issue, but the `{Task, &handler/0}` as a child looks very compelling if it works.

We supervise these Agents using a DynamicSupervisor that we configure [here](https://github.com/elixir-grpc/grpc-reflection/blob/main/lib/grpc_reflection.ex#L117), what you'll likely want to do is to move the DynamicSupervisor there from the local tree root and place it under a normal supervisor -- then you have a place to put your one-time setup job.

So we go from
```mermaid
graph TD;
      App(User Application)-->GrpcReflection;
      GrpcReflection-->DynamicSupervisor;
      DynamicSupervisor-->Agents;
```
to
```mermaid
graph TD;
      App(User Application)-->GrpcReflection;
      GrpcReflection-->Supervisor;
      Supervisor-->InitTask;
      Supervisor-->DynamicSupervisor;
      DynamicSupervisor-->Agents;
```

I hope that makes sense

Reply:
I attempted to put your idea into practice, but I am uncertain if my interpretation of it was accurate.

Reply:
Your interpretation is correct and exactly what I meant.